src/datasets/arrow_writer.py

```diff
@@ -184,8 +184,9 @@ def __arrow_array__(self, type: Optional[pa.DataType] = None):
             elif isinstance(data, list) and data and isinstance(first_non_null_value(data)[1], np.ndarray):
                 out = list_of_np_array_to_pyarrow_listarray(data)
             else:
-                out = pa.array(cast_to_python_objects(data, only_1d_for_numpy=True))
-
+                casted_data = cast_to_python_objects(data, only_1d_for_numpy=True)
+                mask = np.array([value is None for value in casted_data])
+                out = pa.array(casted_data, mask=mask)
```

Comment:
we should keep the null values here no ?

Reply:
Oh maybe I misunderstood: is mask removing the exampels (as in, ignoring them), or simply setting them to None ?

Reply:
I think you are right. Let me remove this part and see if the newly-added tests are still green.

Reply:
Could you explain why you need to specify `mask` here ? Isn't it inferred by pyarrow automatically ?

Reply:
> it indeed makes them null ! nice :)
Yes, but nulls should be preserved here anyway :).